v2/jdbc-to-spanner/src/main/java/com/google/cloud/teleport/v2/spanner/ResultSetToMutation.java

```diff
@@ -0,0 +1,115 @@
+import com.google.cloud.spanner.Mutation;
+import java.sql.Array;
+import java.sql.Clob;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.beam.sdk.io.jdbc.JdbcIO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link JdbcIO.RowMapper} implementation to convert Jdbc ResultSet rows to Cloud Spanner
+ * Mutation.
+ */
+public class ResultSetToMutation implements JdbcIO.RowMapper<Mutation> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ResultSetToMutation.class);
+
+  /** Factory method for {@link ResultSetToMutation}. */
+  public static JdbcIO.RowMapper<Mutation> create(
+      String table, Set<String> columnsToIgnore) {
+    return new ResultSetToMutation(table, columnsToIgnore);
+  }
+
+  private String table;
+  private Set<String> columnsToIgnore;
+
+  private ResultSetToMutation(String table, Set<String> columnsToIgnore) {
+    this.table = table;
+    this.columnsToIgnore = columnsToIgnore == null ? Collections.emptySet() : columnsToIgnore;
+  }
+
+  @Override
+  public Mutation mapRow(ResultSet resultSet) throws Exception {
+    ResultSetMetaData metaData = resultSet.getMetaData();
+    Mutation.WriteBuilder mutation = Mutation.newInsertOrUpdateBuilder(table);
+    for (int i = 1; i <= metaData.getColumnCount(); i++) {
+      Object columnVal = resultSet.getObject(i);
+      if (columnVal == null) {
+        continue;
+      }
+      String columnName = metaData.getColumnName(i);
+      if (columnsToIgnore.contains(columnName)) {
+        continue;
+      }
+      int columnType = metaData.getColumnType(i);
+      switch (columnType) {
+        case java.sql.Types.VARCHAR:
+        case java.sql.Types.CHAR:
+        case java.sql.Types.LONGVARCHAR:
+          mutation.set(columnName).to(String.valueOf(columnVal));
+          break;
+        case java.sql.Types.BIGINT:
+        case java.sql.Types.INTEGER:
+        case java.sql.Types.SMALLINT:
+        case java.sql.Types.TINYINT:
+          mutation.set(columnName).to(Long.valueOf(columnVal.toString()));
+          break;
+        case java.sql.Types.BOOLEAN:
+        case java.sql.Types.BIT:
+          mutation.set(columnName).to(resultSet.getBoolean(i));
+          break;
+        case java.sql.Types.DOUBLE:
+        case java.sql.Types.FLOAT:
+        case java.sql.Types.REAL:
+          mutation.set(columnName).to(((Number) columnVal).doubleValue());
+          break;
+        case java.sql.Types.TIMESTAMP:
+        case java.sql.Types.TIME:
+          com.google.cloud.Timestamp ts =
+              com.google.cloud.Timestamp.of(resultSet.getTimestamp(i));
```

Comment:
Do we need to convert to UTC before storing?

Reply:
Timestamp will be in UTC by default.